### ISSUE TITLE
Run Coati 0.7 on Ubuntu : could not find or load Qt platform plugin

### ISSUE THREAD

**Issue by rojaster** [CoatiSoftware/CoatiBugTracker#97]
run Coati on Ubuntu 15.10 from shell and it was failed
```bash
$<path_to_coati>: ./Coati

This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: xcb.

Reinstalling the application may fix this problem.
Aborted (core dumped)
```

**Comment by rojaster** [comment]
I recognized that i have libs for xcb:

I've been run it via ./Coati.sh 
I think cause in my Qt5 installation path  ~/Qt

Will Try this way on Mac

**Comment by egraether** [comment]
@rojaster Are you able to run Coati 0.7 on Ubuntu now?

**Comment by rojaster** [comment]
Yeap, but from sh script only. Cause it adds some libraries into PATH. 

I have manually installed Qt5 and one Qt(4) installed from repository. And libxbc.so.1 situates in different places. One into my home QT5 path and one into my defaults path

**Comment by rojaster** [comment]
Hey, I have the same issue like on my macOS. I think it is the same. Look **Segmentation fault (core dumped)**

```bash

/usr/lib/gcc/x86_64-linux-gnu/5/include/xmmintrin.h:257:10: error: use of undeclared identifier '__builtin_ia32_xorps'
  return __builtin_ia32_xorps (__A, __B);
         ^
/usr/lib/gcc/x86_64-linux-gnu/5/include/xmmintrin.h:285:19: error: use of undeclared identifier '__builtin_ia32_movss'
  return (__m128) __builtin_ia32_movss ((__v4sf) __A,
                  ^
/usr/lib/gcc/x86_64-linux-gnu/5/include/xmmintrin.h:295:19: error: use of undeclared identifier '__builtin_ia32_movss'
  return (__m128) __builtin_ia32_movss ((__v4sf) __A,
                  ^
/usr/lib/gcc/x86_64-linux-gnu/5/include/xmmintrin.h:323:19: error: use of undeclared identifier '__builtin_ia32_movss'
  return (__m128) __builtin_ia32_movss ((__v4sf) __A,
                  ^
fatal error: too many errors emitted, stopping now
15:45:23 | ParserClientImpl.cpp:559 addComponentAccess() | ERROR: Cannot assign access1 to node id 5286 because it's not a child node.
15:45:23 | ParserClientImpl.cpp:559 addComponentAccess() | ERROR: Cannot assign access1 to node id 5287 because it's not a child node.
15:45:23 | ParserClientImpl.cpp:559 addComponentAccess() | ERROR: Cannot assign access1 to node id 5288 because it's not a child node.
7620 errors generated.
Error while processing /.../obfuscators/ollvm/obfuscator/lib/Target/ARM/Thumb2SizeReduction.cpp.
15:45:24 | ParserClientImpl.cpp:559 addComponentAccess() | ERROR: Cannot assign access1 to node id 2307 because it's not a child node.
7757 errors generated.
Error while processing /..../obfuscators/ollvm/obfuscator/lib/Analysis/LoopInfo.cpp.
**Segmentation fault (core dumped)**


```

**Comment by egraether** [comment]
@rojaster I'm glad you got it to run! No this is different than the MacOSX issue. It looks like there is something wrong with your project setup. The log says that 7757 error where generated. That means Coati received a lot of garbage from the indexer and crashed while trying to make sense of it.

Which type of project setup did you choose? Empty Project or From Compilation Database?

The files you are trying to index have a dependency to the gcc compiler and some symbols are missing:
```
/usr/lib/gcc/x86_64-linux-gnu/5/include/xmmintrin.h:323:19: error: use of undeclared identifier '__builtin_ia32_movss'
  return (__m128) __builtin_ia32_movss ((__v4sf) __A,
```
You most likely didn't set all necessary **Include Paths**, so Coati cannot find all included header files. Have a look at the errors in your console and look for `"<filename>" file not found` errors and then add the path to this file to the **Include Paths** in your project settings.

Or alternatively send us the most recent log files from the [data folder](https://www.coati.io/documentation/#Datafolder) so we can have a closer look.

**Comment by egraether** [comment]
@rojaster One more thing: While fixing the project setup it is better to have just one indexer thread running. Set **Indexer threads** in the [Preferences](https://www.coati.io/documentation/#PreferencesWindow) to 1.

**Comment by rojaster** [comment]
I haven't changed preferences by default =-)

**Comment by rojaster** [comment]
Do u use  Qt, which version is ?

**Comment by egraether** [comment]
Set **Indexer threads** in the [Preferences](https://www.coati.io/documentation/#PreferencesWindow) to 1, so the source files are not indexed in parallel.

We are using Qt 5.6

**Comment by rojaster** [comment]
> The files you are trying to index have a dependency to the gcc compiler and some symbols are missing:

i've added gcc paths when created project

> We are using Qt 5.6
I have Qt5.5 that i've installed manually. May be libxbc depends on from version of Qt?

> Or alternatively send us the most recent log files from the data folder so we can have a closer look.
Ok

**Comment by st4ll1** [comment]
On Linux you need to run Coati via the script. We think about another way to do this, since its confusing to open the application via script instead of open the application direct.

**Comment by rojaster** [comment]
> On Linux you need to run Coati via the script. We think about another way to do this, since its confusing to open the application via script instead of open the application direct.

On linux it is normally

**Comment by egraether** [comment]
Closing, since this was resolved and we have the separate issue #99 for the GCC __builtin problems now.

**Comment by dukebarman** [comment]
I have a same bug:

```
Coati_0_7_0_0_Linux_64bit $ ./Coati
This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: xcb.

Reinstalling the application may fix this problem.
[1]    83843 abort (core dumped)  ./Coati

```

OS: Linux astro 4.4.13-1-MANJARO #1 SMP PREEMPT Wed Jun 8 09:21:59 UTC 2016 x86_64 GNU/Linux
